### PR TITLE
Add libxxhash package

### PR DESCRIPTION
**Pull request by Toasterson**

### CODE REVIEW

File: components/library/libxxhash/Makefile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2024 <contributor>
+#
```
Comment (AndWac) [inline components/library/libxxhash/Makefile:14]: You should enter something here.

File: components/library/libxxhash/Makefile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).zip
+COMPONENT_ARCHIVE_URL= https://github.com/Cyan4973/xxHash/archive/refs/tags/v$(COMPONENT_VERSION).zip
+COMPONENT_ARCHIVE_HASH= sha256:a33d7e8798bebb297095b715f93a71c5e535919434ce27ddc65e72e4e0fda3b9
+COMPONENT_LICENSE= BSD 2-Clause License
+
+TEST_TARGET=$(NO_TESTS) # if no testsuite enabled
+include $(WS_MAKE_RULES)/common.mk
+
+COMPONENT_INSTALL_ENV += PREFIX=/usr
+COMPONENT_INSTALL_ENV += LIBDIR=/usr/lib/amd64
```
Comment (AndWac) [inline components/library/libxxhash/Makefile]: We usually use constants from shared-macros.mk here; $(USRDIR) resp. $(USRLIBDIR64)

Comment (Toasterson) [reply components/library/libxxhash/Makefile]: I got them working.

File: components/library/libxxhash/Makefile (hunk 1 above)
Comment (Toasterson) [reply components/library/libxxhash/Makefile:14]: Oh, right missed this one.

File: components/library/libxxhash/xxhsum.p5m [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2024 Till Wegmüller
+#
+
+set name=pkg.fmri value=pkg:/sysutils/xxhsum@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.human-version value=$(HUMAN_VERSION)
```
Comment (AndWac) [inline components/library/libxxhash/xxhsum.p5m:17]: sysutils?
COMPONENT_FMRI  should be used here!

Comment (Toasterson) [reply components/library/libxxhash/xxhsum.p5m:17]: no I am seperating the binary and library package as the library is used in many applications and the binary is statically compiled thus does not need the library.

Comment (mtelka) [reply components/library/libxxhash/xxhsum.p5m:17]: Please note that you can use `$(COMPONENT_FMRI)` here and set `COMPONENT_FMRI.xxhsum = sysutils/xxhsum` in the `Makefile`.

Comment (Toasterson) [reply components/library/libxxhash/xxhsum.p5m:17]: Oh, thats a magic that i never used before. Switched.

Comment (mtelka) [reply components/library/libxxhash/xxhsum.p5m:17]: Not surprising because it is pretty new and not widely used yet.

**Comment by mtelka** [comment]
@Toasterson, it is not possible to install the `sysutils/xxhsum` package.  Please fix it.  Thank you.
```
# pkg install -v sysutils/xxhsum
          Packages to install:  1
       Create boot environment: No
Create backup boot environment: No
          Rebuild boot archive: No

Changed packages:
openindiana.org
  sysutils/xxhsum
    None -> 0.8.2-2024.0.0.0

pkg install: The following packages deliver conflicting action types to usr/man:

  link:
    pkg://openindiana.org/system/man@0.5.11,5.11-2024.0.0.22092:20240312T012939Z
  dir:
    pkg://openindiana.org/sysutils/xxhsum@0.8.2,5.11-2024.0.0.0:20240311T222639Z

These packages cannot be installed together. Any non-conflicting subset
of the above packages can be installed.
# 
```

**Pull request merged**